src/UI/Component/Panel/Listing/AppointementItem.php

```diff
@@ -0,0 +1,79 @@
+<?php
+
+/* Copyright (c) 2016 Timon Amstutz <timon.amstutz@ilub.unibe.ch> Extended GPL, see docs/LICENSE */
+
+namespace ILIAS\UI\Component\Panel\Listing;
+
+/**
+ * Interface AppointmentItem
+ * @package ILIAS\UI\Component\Panel\Listing
+ */
+interface AppointmentItem extends \ILIAS\UI\Component\Component {
+	
+	/**
+	 * Gets the title of the appointment item
+	 *
+	 * @return string
+	 */
+	public function getTitle();
+
+	/***
+	 * Get the starting point of the appointment.
+	 * @return ilDateTime
+	 */
+	public function getFrom();
+
+	/***
+	 * Get the ending point of the appointment.
+	 * @return ilDateTime
+	 */
+	public function getEnd();
+
+	/***
+	 * Get the color of the calendar containing the item as color code (hex).
+	 * @return string
+	 */
+	public function getColor();
+
+	/**
+	 * Creates a new appointment item with a description.
+	 * @param string $description
+	 * @return AppointmentItem
+	 */
+	public function withDescription($description);
+
+	/**
+	 * Get the description of the appointment.
+	 * @return string
+	 */
+	public function getDescription();
+
+	/**
+	 * Set of meta data as key-value pairs. The key is holding the title and the
+	 * value is holding the content of the specific data set.
+	 * @param [] $meta_data string (Title) => string (Content)
+	 * @return AppointmentItem
+	 */
+	public function withMetaData($meta_data);
```

Comment:
I'm not sure i like the `MetaData` term here. Is this really described correctly as data about data? To me, this MetaData term somehow gotten out of hand =)

This could just be named properties or details.